src/lib.rs

```diff
@@ -0,0 +1,108 @@
+pub mod config {
+    use serde::Deserialize;
+
+    use std::error::Error;
+    use std::fs::File;
+    use std::io::BufReader;
+    use std::path::Path;
+
+    #[derive(Deserialize, Debug)]
+    pub struct Config {
+        pub db_address: String,
+    }
+
+    pub fn read_json_config(path: impl AsRef<Path>) -> Result<Config, Box<dyn Error>> {
+        let file = File::open(path)?;
+        let reader = BufReader::new(file);
+
+        // read json from file
+        let json_data = serde_json::from_reader(reader)?;
+
+        Ok(json_data)
+    }
+}
+
+pub mod db {
+    use chrono::{serde::ts_seconds, serde::ts_seconds_option, DateTime, Utc};
+    use mongodb::{bson::doc, sync::Collection};
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Debug, Serialize, Deserialize)]
+    // Task型
+    pub struct Task {
+        #[serde(with = "ts_seconds_option")]
+        pub scheduled_date: Option<chrono::DateTime<Utc>>,
+        #[serde(with = "ts_seconds")]
+        pub register_date: DateTime<Utc>,
+        pub title: String,
+        pub memo: String,
+        pub finish: bool,
+    }
+
+    impl Default for Task {
+        fn default() -> Self {
+            Self {
+                scheduled_date: None,
+                register_date: Utc::now(),
+                title: "".into(),
+                memo: "".into(),
+                finish: false,
+            }
+        }
+    }
+
+    //新規タスクの追加
+    pub fn add_task(
+        new_task: Task,
+        collection: &Collection<Task>,
+    ) -> Result<(), mongodb::error::Error> {
+        collection.insert_one(new_task, None)?;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        config::{read_json_config, Config},
+        db::{add_task, Task},
+    };
+    use chrono::{TimeZone, Utc};
```

Comment:
```suggestion
    use chrono::Utc;
```

Reply:
>Makes a new Date from year, month, day and the current time zone. This assumes the proleptic Gregorian calendar, with the year 0 being 1 BCE.
>The time zone normally does not affect the date (unless it is between UTC-24 and UTC+24), but it will propagate to the DateTime values constructed via this date.
>
>Panics on the out-of-range date, invalid month and/or day.

>[Example](https://docs.rs/chrono/0.3.0/chrono/offset/trait.TimeZone.html#example)
```rs
use chrono::{UTC, TimeZone};
```

らしいので、TimeZoneが必須って認識でいいんですかね？

Reply:
あ〜〜〜 `TimeZone` ってtraitか, これは失礼しました
Exampleへのリンクくそ古いのだけ気になりましたが… ちゃんとversionに合ったdocを読んでいますか？ (このPjで使われているchronoは [**`0.4.19`**](https://docs.rs/chrono/0.4.19/chrono/offset/trait.TimeZone.html#example) です…)

Reply:
docをどこで読めばいいかを実はちゃんと理解してなかった。
[crates.io](https://crates.io)じゃなくて[docs](https://docs.rs)で読めばいいって認識であってますか？

Reply:
https://docs.rs がdocsの置かれてる場所なので, 何も考えずにアドレスバーに `docs.rs/<crate_name>` と打ち込みましょう latestのdocへのaliasになってます
ちなみに標準ライブラリにも繋がってます, ほら: https://docs.rs/std

Reply:
ありがとうございます。